src/sentry/eventstream/kafka/backend.py

```diff
@@ -128,14 +129,19 @@ def insert(
         group_states: Optional[GroupStates] = None,
         **kwargs,
     ):
-        message_type = "transaction" if self._is_transaction_event(event) else "error"
+        event_type = self._get_event_type(event)
 
-        if message_type == "transaction" and self.assign_transaction_partitions_randomly:
+        if event_type == EventStreamEventType.Generic:
+            assign_partitions_randomly = True
+        elif (
```

Comment:
worth turning into an `or` for one logical eval? just a style comment not opinionated about it